bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/handler/MiIoVacuumHandler.java

```diff
@@ -131,16 +131,18 @@ private boolean updateVacuumStatus(JsonObject statusData) {
         updateState(CHANNEL_CLEAN_TIME,
                 new DecimalType(TimeUnit.SECONDS.toMinutes(statusData.get("clean_time").getAsLong())));
         updateState(CHANNEL_DND_ENABLED, new DecimalType(statusData.get("dnd_enabled").getAsBigDecimal()));
-        updateState(CHANNEL_ERROR_CODE,
-                new StringType(VacuumErrorType.getType(statusData.get("error_code").getAsInt()).getDescription()));
+        VacuumErrorType errorCode = VacuumErrorType.getType(statusData.get("error_code").getAsInt());
+        updateState(CHANNEL_ERROR, new StringType(errorCode.getDescription()));
+        updateState(CHANNEL_ERROR_CODE, new DecimalType(statusData.get("error_code").getAsInt()));
         int fanLevel = statusData.get("fan_power").getAsInt();
-        FanModeType fanpower = FanModeType.getType(fanLevel);
+        FanModeType fanPower = FanModeType.getType(fanLevel);
```

Comment:
Same here: why is thsi a local variable (and other states are not)?

Reply:
You are right, this makes no sense, I just made a small camel case changes. I just took it from the original code.